src/lib/net/dns.ts

```diff
@@ -0,0 +1,61 @@
+////////////////////////////////////////////////////////////////////////////////
+// DNS helper functions
+//
+// Use these functions for sanitization and validation of DNS names and DNS name
+// segments.
+////////////////////////////////////////////////////////////////////////////////
+
+export const validDnsRegex = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$/;
```

Comment:
Do we need to check for uppercase characters? A-Z

Reply:
Although RFC1123 DNS spec says DNS names are case sensitive, k8s DNS only supports lower case (that regex is actually ripped straight from k8s DNS core validator)

Reply:
ok